packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3662,6 +3679,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     // unfocused field that previously had a selection in the same spot.
     if (value == textEditingValue) {
       if (!widget.focusNode.hasFocus) {
+        _nextFocusChangeIsInternal = true;
         widget.focusNode.requestFocus();
```

Comment:
Do you have to handle the case where something else takes the focus between the time that this focus request is sent and the time that the focus change gets called?

It's possible that the operations could go like this:

 - `_nextFocusChangeIsInternal` is set to true.
 - text field calls requestFocus
 - before the microtask that resolves focus executes, something else calls request focus, superseding the text field's focus request.
 - Later on, the text field again gets focus, but from an external source.
 
At that point, `_nextFocusChangeIsInternal` would be true, but the focus change wouldn't have been internal.

That's why I was suggesting that maybe the text field needs to listen to the `FocusManager` and if any focus changes happen that don't involve `widget.focusNode` receiving focus, then you'd reset `_nextFocusChangeIsInternal`.

Reply:
Ah sorry, I see what you mean.  I've gone ahead with this plan of listening to FocusManager.  I also added a test for the scenario you described in your bullet points.  Let me know if you think that should cover this case.